src/settings/DesktopUpdateHelpLabel.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+import m from "mithril"
+import {Icon} from "../gui/base/Icon"
+import {noOp} from "../api/common/utils/Utils"
+import {lang} from "../misc/LanguageViewModel"
+import {BootIcons} from "../gui/base/icons/BootIcons"
+import {nativeApp} from '../native/NativeWrapper.js'
+import {Request} from "../api/common/WorkerProtocol.js"
+
+export type UpdateHelpLabelAttrs = {
+	updateAvailable: Stream<boolean>;
+}
+
+export class DesktopUpdateHelpLabel {
+	_waiting: boolean;
+	_error: boolean;
+
+	getActionLink(updateAvailable: Stream<boolean>): Child {
+		if (this._waiting || this._error) return null
+		return m("span.text-break.pr-s", m('button.underline', {
+				type: "button",
+				href: "#",
+				tabindex: "0",
+				role: "button",
+				onclick: this.getOnClick(updateAvailable)
+			}, lang.get(updateAvailable() ? "installNow_action" : "checkAgain_action"))
+		)
+	}
+
+	getLabel(updateAvailable: Stream<boolean>): Child {
+		let ret = ""
+		if (updateAvailable()) {
+			ret = lang.get("updateFound_label")
+		} else if (this._error) {
+			ret = lang.get("serviceUnavailable_msg")
+		} else if (this._waiting) {
+			ret = lang.get("checkingForUpdate_action")
+		} else {
+			ret = lang.get("noUpdateAvailable_msg")
+		}
+		return m("span.pr-s", ret + " ")
+	}
+
+	getIcon(): Child {
+		return this._waiting && !this._error
+			? m(Icon, {
+				icon: BootIcons.Progress,
+				class: 'flex-center items-center icon-progress-tiny icon-progress'
+			})
+			: null
+	}
+
+	getOnClick(updateAvailable: Stream<boolean>): () => void {
```

Comment:
I think it's odd to decide which function to return, I would rather add `onClick()` method and decided what to do inside of it when the click actually happens.